### ISSUE TITLE
[FEATURE REQUEST] Alternative to the full scan of user files

### ISSUE THREAD

**Issue by diocas** [owncloud/ios-app#774]
As discussed on Thursday, I'm opening this issue so that we can follow up on this subject. As agreed, this needs to be discussed further.

## Problem description
At CERN we provide a home directory to all of our users with 1TB and 1M inodes limit. Users also have access to Projects, that can have much higher limits.
Currently, the iOS client does a PROPFIND for each file the user has. Besides that, and unlike the Desktop Sync Client where adding a shared folder is done manually, when visiting the tab "shared with me" it also starts PROPFIND'ing other users/projects spaces. This can lead to millions of PROPFIND's to the server and a possible impact on the performance of both the client and the server.
This feature might also be unpractical in slow/unreliable connections, which are more common in mobile devices.

(as a side note, this amount of files should also be taken in consideration when doing performance tests on the app)

## Possible solution
At this point, a solution is not clear and needs to be discussed. Maybe an option to disable this behavior would be good.

If the reason this is done in the first place is just so that the app can provide special functionalities - like the quick access menu -, maybe we could do it on the server side and use the computed values by all the clients, like iOS, Android and even Phoenix.

**Comment by diocas** [comment]
@michaelstingl Let's try and see it. As we agreed, I will create an internal ticket for a branded version.

**Comment by michaelstingl** [comment]
> Currently, the iOS client does a PROPFIND for each file the user has

This is not the case. iOS app does a PROPFIND only for the folder, NOT for every file in the folder (similar desktop app)

**Comment by diocas** [comment]
We're running tests on the branded beta app and the problem is that on the web we expose shares inside a hidden "__myshares" folder and projects inside "__myprojects". So, when we access the quick access menu, eventually the app starts propfinding inside those folders (which was what I observed earlier). And the idea was to have something similar in OCIS.

So it would be great to be able to disable this, which would mean disabling many of the quick access functionality (or, as I suggested, rely on the server for the functionality).
